src/client.rs

```diff
@@ -1282,14 +1282,112 @@ impl<T: Read + Write> Session<T> {
         .and_then(|lines| parse_id_seq(&lines, &mut self.unsolicited_responses_tx))
     }
 
+    /// The [`SETACL` command](https://datatracker.ietf.org/doc/html/rfc4314#section-3.1)
+    ///
+    /// Modifies the ACLs on the given mailbox for the specified identifier.
+    /// Return [`Error::No`] if the logged in user does not have `a` rights on the mailbox.
+    ///
+    /// This method only works against a server with the ACL capability. Otherwise [`Error::Bad`]
```

Comment:
Is "ACL" the actual name for the capability one should check for on the server? Or does it have some other name that'll be what has to be checked for?

Reply:
ACL is the actual name in the server.n (So `has_str("ACL")`)

There are other "sub-pieces" for determining additional functionality. e.g.
- checking `RIGHTS=kx`
   - which the imap-proto crate does not easily allow you to query as `RIGHTS=` lists extra rights characters supported by the server.  I'll have to do a new PR so support that one. (and make the Capability enum non_exhaustive)
- also the ANNOTATE-EXPERIMENT-1 capability to detect the annotation extension (adds the `n` ACL)